cli/flox/src/commands/environment.rs

```diff
@@ -1696,7 +1696,7 @@ impl Pull {
         force: bool,
         message: &str,
     ) -> Result<()> {
-        if dot_flox_path.exists() {
+        if dot_flox_path.exists() && !force {
```

Comment:
suggestion: I think we should make sure the old environment is cleaned up - there could be other files in `.flox`, and there could be other files living in FLOX_CACHE_DIR. I think cleaning up would look like trying to `open()` the `.flox` directory. If that succeeds, call `Environment::delete()`. If it fails, just remove the directory (or make sure somehow that there aren't any files leftover that we don't overwrite in the rest of the method

Reply:
> suggestion: I think we should make sure the old environment is cleaned up - there could be other files in `.flox`, and there could be other files living in FLOX_CACHE_DIR. I think cleaning up would look like trying to `open()` the `.flox` directory. If that succeeds, call `Environment::delete()`. If it fails, just remove the directory (or make sure somehow that there aren't any files leftover that we don't overwrite in the rest of the method)

Good suggestion, thanks! I ended up using `ManagedEnvironment::open(...)`. I'm not 100% percent sure about it though.

Reply:
I led you astray sorry it should actually be `open_path()` - that calls `DotFlox::open()` and then `into_concrete_environment()` which could end up calling either `ManagedEnvironment::open()` or `PathEnvironment::open()`. `open()` was probably not specific enough 😳 

Basically a `.flox` directory could either contain an environment created with `flox init`, or it could contain one created by `flox pull`. This should be able to handle both of them.

Reply:
Before committing it, how's this?

```rust
        if dot_flox_path.exists() {                                                                                                                                                                                                                            
            if force {                                                                                                                                                                                                                                         
                match open_path(flox, &dot_flox_path) {                                                                                                                                                                                                        
                    Ok(concrete_env) => match concrete_env {                                                                                                                                                                                                   
                        ConcreteEnvironment::Path(env) => {                                                                                                                                                                                                    
                            env.delete(flox)                                                                                                                                                                                                                   
                                .context("Failed to delete existing environment")?;                                                                                                                                                                            
                        },                                                                                                                                                                                                                                     
                        ConcreteEnvironment::Managed(env) => {                                                                                                                                                                                                 
                            env.delete(flox)                                                                                                                                                                                                                   
                                .context("Failed to delete existing environment")?;                                                                                                                                                                            
                        },                                                                                                                                                                                                                                     
                        ConcreteEnvironment::Remote(_) => { },                                                                                                                                                                                                                                     
                    },                                                                                                                                                                                                                                         
                    Err(_) => {                                                                                                                                                                                                                                
                        fs::remove_dir_all(&dot_flox_path).context(format!(                                                                                                                                                                                    
                            "Failed to remove existing .flox directory at {:?}",                                                                                                                                                                               
                            dot_flox_path                                                                                                                                                                                                                      
                        ))?;                                                                                                                                                                                                                                   
                    },                                                                                                                                                                                                                                         
                }                                                                                                                                                                                                                                              
            } else {                                                                                                                                                                                                                                           
                bail!(                                                                                                                                                                                                                                         
                    "An environment already exists at {:?}. Use --force to overwrite.",                                                                                                                                                                        
                    dot_flox_path                                                                                                                                                                                                                              
                );                                                                                                                                                                                                                                             
            }                                                                                                                                                                                                                                                  
        }                                                                                                                                                                                                                                                      
```

Reply:
Looks great!

Reply:
Thanks, committed!